flutter/lib/src/event_processor/platform_exception_event_processor.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:sentry/sentry.dart';
+
+/// Add code & message from [PlatformException] to [SentryException]
+class PlatformExceptionEventProcessor implements EventProcessor {
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) {
+    final exceptions = <SentryException>[];
+
+    for (SentryException exception in (event.exceptions ?? [])) {
+      final platformException = exception.throwable;
+      if (platformException is PlatformException) {
+        exception = _enrich(exception, platformException);
+      }
+      exceptions.add(exception);
+    }
+
+    return event.copyWith(exceptions: exceptions);
+  }
+
+  SentryException _enrich(
+      SentryException sentryException, PlatformException platformException) {
+    final data = Map<String, dynamic>.from(
+      sentryException.mechanism?.data ?? {},
+    );
+    data['platformException'] = {'code': platformException.code};
+    if (platformException.message != null) {
+      data['platformException']['message'] = platformException.message;
+    }
+    return sentryException.copyWith(
+      mechanism: sentryException.mechanism?.copyWith(data: data),
+    );
```

Comment:
This will only work if the event has a `Mechanism` which is not the case for manually captured events.
Maybe if there's not one, we create a new mechanism?